OpenEdXMobile/src/main/java/org/edx/mobile/view/view_holders/DiscussionSocialLayoutViewHolder.java

```diff
@@ -41,26 +41,68 @@ public DiscussionSocialLayoutViewHolder(View itemView) {
     }
 
     public void setDiscussionThread(final DiscussionThread discussionThread) {
-        threadVoteTextView.setText(ResourceUtil.getFormattedStringForQuantity(
-                threadVoteTextView.getResources(), R.plurals.discussion_responses_action_bar_vote_text, discussionThread.getVoteCount()));
-        threadVoteIconImageView.setIconColorResource(discussionThread.isVoted() ?
-                R.color.edx_brand_primary_base : R.color.edx_brand_gray_base);
-
+        setVote(discussionThread.isVoted(), discussionThread.isVoted() ? discussionThread.getVoteCount()-1: discussionThread.getVoteCount());
+        setFollowing(discussionThread.isFollowing());
         threadFollowContainer.setVisibility(View.VISIBLE);
+    }
+
+    public void setDiscussionResponse(final DiscussionComment discussionResponse) {
+        setVote(discussionResponse.isVoted(), discussionResponse.isVoted() ? discussionResponse.getVoteCount()-1: discussionResponse.getVoteCount());
+    }
 
-        if (discussionThread.isFollowing()) {
+    public boolean toggleFollow(){
+        setFollowing(!threadFollowContainer.isSelected());
+        return threadFollowContainer.isSelected();
+    }
+
+    public boolean toggleVote(int otherUserVotes){   //OtherUserVotes is the amount of votes on the post not counting this users post
```

Comment:
The comment here could be moved into a proper javadoc for this function instead.